src/components/ConnectToXeroButton/index.tsx

```diff
@@ -50,6 +64,17 @@ function ConnectToXeroButton({policyID, shouldDisconnectIntegrationBeforeConnect
                     danger
                 />
             )}
+            {isRequire2FAModalOpen && (
+                <RequireTwoFactorAuthenticationModal
+                    onSubmit={() => {
+                        setIsRequire2FAModalOpen(false);
+                        Navigation.navigate(ROUTES.SETTINGS_2FA.getRoute(ROUTES.POLICY_ACCOUNTING.getRoute(policyID), getXeroSetupLink(policyID)));
```

Comment:
```suggestion
Navigation.dismissModal();                    Navigation.navigate(ROUTES.SETTINGS_2FA.getRoute(ROUTES.POLICY_ACCOUNTING.getRoute(policyID), getXeroSetupLink(policyID)));
```

I think for narrow layout, another `FullScreenNavigator` is being pushed here because the top navigator is `RightHandModalNavigator`. Dismissing RHP before navigating fixes it. Same for `index.native.tsx`.

Reply:
@rushatgabhane

Reply:
Not convinced by this? @rushatgabhane

Reply:
@c3024 sorry, let me give this a try

Reply:
this works well!